homeassistant/components/evohome.py

```diff
@@ -0,0 +1,606 @@
+            CONF_LOCATION_IDX
+        )
+
+        domain_data['params'][CONF_LOCATION_IDX] = 0
+        domain_data['config'] = client.installation_info[0]
+
+    domain_data['status'] = {}
+
+    if _LOGGER.isEnabledFor(logging.DEBUG):
+        _LOGGER.debug(
+            "setup(): Location/TCS (temp. control system) used is: %s [%s]",
+            domain_data['config'][GWS][0][TCS][0]['systemId'],
+            domain_data['config']['locationInfo']['name']
+        )
+        # Some platform data needs further redaction before being logged.
+        tmp = dict(domain_data['config'])
+        tmp['locationInfo']['postcode'] = 'REDACTED'
+
+        _LOGGER.debug("setup(): domain_data['config']: %s", tmp)
+
+    # We have the platofrom configuration, but no state as yet, so...
+    def _first_update(event):
+        """Let the controller know it can obtain it's first update."""
+    # Send a message to the hub to do its first update()
+        pkt = {
+            'sender': 'setup()',
+            'signal': 'update',
+            'to': EVO_MASTER
+        }
+        hass.helpers.dispatcher.dispatcher_send(
+            DISPATCHER_EVOHOME,
+            pkt
+        )
+
+    hass.bus.listen(EVENT_HOMEASSISTANT_START, _first_update)
+
+    load_platform(hass, 'climate', DOMAIN)
+
+    return True
+
+
+class EvoEntity(ClimateDevice):
+    """Base for Honeywell evohome slave devices (Heating/DHW zones)."""
+
+    def __init__(self, hass, client, obj_ref):
+        """Initialize the evohome entity.
+
+        Most read-only properties are set here.  So are pseudo read-only,
+        for example name (which _could_ change between update()s).
+        """
+        # Set the usual object references
+        self.hass = hass
+        self.client = client
+        domain_data = hass.data[DATA_EVOHOME]
+
+        # Set the entity's own object reference & identifier
+        self._id = obj_ref.systemId
+
+        # Set the entity's configuration shortcut (considered static)
+        self._config = domain_data['config'][GWS][0][TCS][0]
+        self._params = domain_data['params']
+
+        # Set the entity's name & icon (treated as static vales)
+        self._name = domain_data['config']['locationInfo']['name']
+        self._icon = "mdi:thermostat"
+
+        # Set the entity's supported features
+        self._supported_features = SUPPORT_OPERATION_MODE | SUPPORT_AWAY_MODE
+
+        # Set the entity's operation list - hard-coded for a particular order,
+        # instead of using self._config['allowedSystemModes']
+        self._op_list = TCS_MODES
+
+        # Create timers, etc. - they're maintained in update(), or schedule()
+        self._status = {}
+        self._timers = domain_data['timers']
+
+        self._timers['statusUpdated'] = datetime.min
+        domain_data['schedules'] = {}
+
+        # Set the entity's (initial) behaviour
+        self._available = False  # will be True after first update()
+        self._should_poll = True
```

Comment:
This should be static as polling is only turned on/off when an entity is added. It might be picked up later if it starts returning True but there is no guarantee.

Reply:
Other entities will be `self._should_poll = False`, you just wont see them until a later PR.

Reply:
removed